src/known_hosts_fmt_plug.c

```diff
@@ -49,7 +49,7 @@ john_register_one(&fmt_known_hosts);
 #define PAD_SIZE                64
 #define BINARY_ALIGN            4
 #define SALT_SIZE               sizeof(struct custom_salt)
-#define SALT_ALIGN              1
+#define SALT_ALIGN              sizeof(unsigned int)
 #define MIN_KEYS_PER_CRYPT      1
 #define MAX_KEYS_PER_CRYPT      1
```

Comment:
```c
static void set_salt(void *salt)
{
        cur_salt = (struct custom_salt *)salt;
}
```
Shouldn't salt alignment be `sizeof(struct custom_salt*)`?

Reply:
I am not able to reproduce this crash again. Should I just drop this PR?

Reply:
No. Due to how set_salt works, salt align should be sizeof pointer, right? So change it to `sizeof(struct custom_salt*)`

Reply:
Thanks. I have updated the PR.

Reply:
I created a wiki page (brain dump) over this issue (and binary align issue).

https://github.com/magnumripper/JohnTheRipper/wiki/Salt-alignment----Binary-alignment.

Most ppl have a better grasp on binary align, because they have worked on CPUs that do force alignment (the SIMD stuff).   But the salt works just the same.  When you allocate, you have to align the buffer to the proper location, THEN all elements in that buffer (when viewed as a structure by the compiler WILL be in the right place).  That rule is to align to element with the largest alignment requirements WITHIN the salt structure.  plain and simple.